src/main/java/com/sciamus/contractanalyzer/domain/checks/queues/kafka/KafkaTransformationCheck.java

```diff
@@ -84,7 +84,8 @@ private String getCheckUniqueIdentifier() {
     private Iterable<ConsumerRecord<String, String>> waitForRecordFromOutsideSystem(String incomingTopic, Consumer<String, String> consumer) {
 
         Iterable<ConsumerRecord<String, String>> records = consumer.poll(Duration.ofSeconds(10)).records(incomingTopic);
-        consumer.close();
+        Try.run(consumer::close)
```

Comment:
## warn

almost ;) What I meant is that if we need to close consumer on any circumstance, then we should use try with finally (`andFinally(Runnable runnable)`). Just to be sure the consumer will be closed even if line 86 will blows up.